rust/lance-index/src/vector/ivf/shuffler.rs

```diff
@@ -78,11 +86,21 @@ impl IvfShuffler {
             pq_width,
             output_dir,
             schema,
+            unsorted_buffers: vec![],
         })
     }
 
+    /// Set the unsorted buffers to be shuffled.
+    ///
+    /// # Safety
+    ///
+    /// user must ensure the buffers are valid.
+    pub unsafe fn set_unsorted_buffers(&mut self, unsorted_buffers: Vec<String>) {
```

Comment:
Why is it a unsafe method? 

Btw, Can we just hold a reference to `&[String]`?

Reply:
This is meant to note to user that if invalid buffers are passed here it can cause undefined behavior

Reply:
changed it to `&[impl Into<String>]`